### DISCUSSION TITLE
Event-Driven API: A Proposal

### DISCUSSION THREAD

**Discussion by TGWaffles** [HypixelDev/PublicAPI#613] [Ideas]
As I'm sure we all know, there's a lot of people playing on Hypixel and a lot of things happening all the time. Cached data turns stale quickly and, as Connor said, the API is a huge project with a large cost.

I'd like to propose a solution that kills 3 birds with one stone: expand the API's use cases, speed up existing applications, and **dramatically reduce the API cost on Hypixel's systems**.

My proposal for this is an event-driven API system that delivers server-side events to listening applications, with near-zero database load and high scalability. This system has already been adopted by numerous high-volume, developer-focused applications, such as Google, YouTube, and Discord.

[An overview is available on the forums](https://hypixel.net/threads/event-driven-api-a-proposal.5492776/) and I've also provided [an implementation outline](https://tgwaffles.notion.site/Hypixel-Event-Based-API-6ccec611c4c84f67bb048ca75a027167) going more in-depth on what this would look like, and how it could be done efficiently. If implemented, this could open the door to countless new opportunities for developers while cutting API costs by orders of magnitude.

**Comment by socalledtheraven** [comment]
Looks like a great idea, excited to see where this goes!

**Comment by aesthetic0001** [comment]
please add this <3

**Comment by Xeravax** [comment]
Someone explained the new API below, probably easier to follow for all the people who don't understand what was being said there.

https://www.youtube.com/watch?v=V-3sCc3ZgnI
